src/toil/job.py

```diff
@@ -1234,7 +807,7 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=None):
                     # either the file was successfully downloaded and added to cache, or something
                     # failed. To prevent code duplication, we recursively call readGlobalFile.
                     flock(lockFileHandle, LOCK_UN)
-                    return self.readGlobalFile(fileStoreID, userPath, cache)
+                    return self.readGlobalFile(fileStoreID, userPath, cache, mutable)
```

Comment:
Optional arguments must always be called by name. I hope you don't do this all over the place!

The reason is that optional arguments should be order-insensitive. When modifying a function, I should be free to reorder the optional arguments with affecting existing call sites of that function. PyCharm doesn't warn you about this?